scripts/west_commands/runners/t32.py

```diff
@@ -0,0 +1,143 @@
+
+import argparse
+import os
+import platform
+import subprocess
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import List, Optional
+
+from runners.core import BuildConfiguration, RunnerCaps, RunnerConfig, ZephyrBinaryRunner
+
+DEFAULT_T32_CONFIG = Path('config.t32')
+
+
+class TRACE32BinaryRunner(ZephyrBinaryRunner):
+    '''
+    Runner front-end for Lauterbach TRACE32.
+
+    The runner is a wrapper around Lauterbach TRACE32 PowerView. It executes a Lauterbach Practice
+    script (.cmm) after launching the debugger, which should be located at
+    zephyr/boards/<board>/support/<command>.cmm, where <board> is the board directory and <command>
+    is the name of the west runner command executed (e.g. flash or debug). Extra arguments can be
+    passed to the startup script by using the command line option --startup-args.
+    '''
+
+    def __init__(self,
+                 cfg: RunnerConfig,
+                 t32_cfg: Path,
+                 arch: str,
+                 startup_args: List[str] = None,
+                 timeout: int = 60) -> None:
+        super(TRACE32BinaryRunner, self).__init__(cfg)
+        self.arch = arch
+        self.t32_cfg = t32_cfg
+        self.t32_exec: Optional[Path] = None
+        self.startup_dir = Path(cfg.board_dir) / 'support'
+        self.startup_args = startup_args
+        self.timeout = timeout
+
+    @classmethod
+    def name(cls) -> str:
+        return 't32'
+
+    @classmethod
+    def capabilities(cls) -> RunnerCaps:
+        return RunnerCaps(commands={'flash', 'debug'})
+
+    @classmethod
+    def do_add_parser(cls, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument('--arch',
+                            default='auto',
+                            choices=('auto', 'arm', 'arm64', 'riscv', 'xtensa'),
+                            help='Target architecture. Set to "auto" to select the architecture '
+                                 'based on CONFIG_ARCH value')
+        parser.add_argument('--config',
+                            default=DEFAULT_T32_CONFIG,
+                            type=Path,
+                            help='Override TRACE32 configuration file path. Can be a relative path '
+                                 'to T32_DIR environment variable, or an absolute path')
+        parser.add_argument('--startup-args',
+                            nargs='*',
+                            help='Arguments to pass to the start-up script')
+        parser.add_argument('--timeout',
+                            default=60,
+                            type=int,
+                            help='Timeout, in seconds, of the flash operation')
+
+    @classmethod
+    def do_create(cls, cfg: RunnerConfig, args: argparse.Namespace) -> 'TRACE32BinaryRunner':
+        build_conf = BuildConfiguration(cfg.build_dir)
+        if args.arch == 'auto':
+            arch = build_conf.getboolean('CONFIG_ARCH').replace('"', '')
```

Comment:
getboolean?

Reply:
Nice catch. It was working without problems due to the implementation of `.getboolean`, changed to `.get`.